app/controllers/schemes_controller.rb

```diff
@@ -25,12 +26,77 @@ def locations
     @total_count = @scheme.locations.size
   end
 
-private
+  def new
+    @scheme = Scheme.new
+  end
+
+  def create
+    @scheme = Scheme.new(clean_params)
+    @scheme.save
+
+    redirect_to edit_scheme_path(id: @scheme.id)
+  end
+
+  def edit
+    if !params[:scheme]
+      @scheme = Scheme.find(params[:id])
+      render "schemes/primary_client_group"
+    elsif primary_client_group_patch?
+      required_params = params.require(:scheme).permit(:primary_client_group)
+      @scheme.update(required_params)
+      # render "schemes/secondary_client_group"
+      render "schemes/check_answers"
+    elsif secondary_client_group_patch?
+      required_params = params.require(:scheme).permit(:secondary_client_group)
+      @scheme.update(required_params)
+      # render "schemes/support"
+      render "schemes/check_answers"
+    elsif support_patch?
+      required_params = params.require(:scheme).permit(:intended_stay, :support_type)
+      @scheme.update(required_params)
+      # render "schemes/details"
+      render "schemes/check_answers"
+    elsif details_patch?
+      required_params = params.require(:scheme).permit(:service_name, :sensitive, :organisation_id, :scheme_type, :registered_under_care_act, :total_units, :id)
+      required_params[:sensitive] = required_params[:sensitive].to_i
+      @scheme.update(required_params)
+      render "schemes/check_answers"
+    end
+  end
+
+  private
+
+  def primary_client_group_patch?
+    params[:scheme][:primary_client_group]
+  end
+
+  def secondary_client_group_patch?
+    params[:scheme][:secondary_client_group]
+  end
+
+  def support_patch?
+    params[:scheme][:intended_stay]
+  end
+
+  def details_patch?
+    params[:scheme][:service_name]
+  end
+
+  def clean_params
+    code = "S#{SecureRandom.alphanumeric(5)}".upcase
```

Comment:
Any reason not to just use the auto incrementing ID here rather than introducing a new gem?

Reply:
This didnt occur to me at the time. And I wanted something that will generate me some IDs quickly so I can focus on other bits, will be more than happy to change this.

Reply:
Since we have to prefix our code with 'S' - will need to write logic for this